### PR TITLE
test: CustomerServiceTest.java

### PR DESCRIPTION
**Pull request by nirinaHelena**

### CODE REVIEW

File: src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    verify(repositoryMock, never()).save(any(Customer.class));
+  }
+
+  @Test
+  void testUpdateCustomersLocationSuccessfulUpdate() {
+    var geoPosition = mock(GeoPosition.class);
+    var coordinates = mock(GeoUtils.Coordinate.class);
+    var customer = mock(Customer.class);
+
+    when(customer.getId()).thenReturn("1");
```
Comment (nvagno) [inline src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java]: No. Anti-pattern. Why mocking when you can create the object?

**Review by nvagno — CHANGES_REQUESTED** [review]

File: src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    subject.updateCustomersLocation();
+    verify(customer, times(1)).setLocation(any(Location.class));
+    verify(repositoryMock, times(1)).save(customer);
+  }
+
+  @Test
+  void testUpdateCustomersLocationInvalidAddress() {
+    var customer = mock(Customer.class);
+
+    when(customer.getId()).thenReturn("1");
```
Comment (nvagno) [inline src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java]: same

**Review by nvagno — CHANGES_REQUESTED** [review]

File: src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java (hunk 1 above)
Comment (nirinaHelena) [reply src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java]: This variable needs to be mocked, as verify is used only with mocks.

File: src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  }
+
+  Customer.CustomerBuilder customerBuilder() {
+    return Customer.builder().id(JOE_DOE_ID);
+  }
+
+  @Test
+  void testUpdateCustomersLocationAddressTooShort() {
+    var customer = customerBuilder().address("12").build();
+
```
Comment (Dinasoa) [inline src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java]: Remove extra blank lines and respect the 'given-when-then' when escaping lines.

File: src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  @Test
+  void testUpdateCustomersLocationSuccessfulUpdate() {
+    var geoPosition = mock(GeoPosition.class);
+    var coordinates = mock(GeoUtils.Coordinate.class);
+    var customer = mock(Customer.class);
+
+    when(customer.getId()).thenReturn("1");
+    when(customer.getFullAddress()).thenReturn("123 Main St, Springfield");
+    when(customer.getAddress()).thenReturn("123 Main St, Springfield");
```
Comment (Dinasoa) [inline src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java]: Do not mock customer, create a Customer object instead and mock the method to return it instead

File: src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    verify(repositoryMock, times(1)).save(customer);
+  }
+
+  @Test
+  void testUpdateCustomersLocationInvalidAddress() {
+    var customer = mock(Customer.class);
+
+    when(customer.getId()).thenReturn("1");
+    when(customer.getFullAddress()).thenReturn("12");
+    when(customer.getAddress()).thenReturn("12");
```
Comment (Dinasoa) [inline src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java]: Same here

**Review by Dinasoa — CHANGES_REQUESTED** [review]

**Pull request closed without merging**